scripts/functions/checksum

```diff
@@ -201,17 +201,27 @@ __rvm_checksum_read()
     [[ -n "${_checksum_md5:-}" ]] ||
       _checksum_md5="$(    __rvm_db_ "$rvm_user_path/md5"      "$_name" | \command \head -n 1 )"
 
-    [[ -n "${_checksum_md5:-}" && $_name == http* ]] ||
-       _checksum_md5="$(__rvm_curl -s -L $_name.md5)"
-
     # sha512
     _checksum_sha512="$(   __rvm_db_ "$rvm_path/config/sha512" "$_name" | \command \head -n 1 )"
 
     [[ -n "${_checksum_sha512:-}" ]] ||
       _checksum_sha512="$( __rvm_db_ "$rvm_user_path/sha512"   "$_name" | \command \head -n 1 )"
 
-    [[ -n "${_checksum_sha512:-}" && $_name == http* ]] ||
-       _checksum_sha512="$(__rvm_curl -s -L $_name.sha512)"
+    __rvm_checksum_none || return 0
```

Comment:
this will never get to checking URL's if there was no checksum locally, `__rvm_checksum_none` has to be moved to the end of the function, outside of loops

Reply:
Good catch, Bash conditions are hard to read.
```bash
__rvm_checksum_none()
{
  [[ -z "${_checksum_md5:-}" && -z "${_checksum_sha512:-}" ]]
}
```
So that returns "true" if both are unset, i.e, it returns "false" if any is set and therefore `|| return 0` returns if any is set.
This is also done at the function entry BTW.

So that seems incorrect in the case we get one of the checksum, but not the other, right?

I think we should return only if both are set then, i.e., replace that with `__rvm_checksum_all && return 0`. I'll do that.

Reply:
@mpapis Should this function return 0 or 1 if we found only one of the checksums?
`__rvm_checksum_validate_file` seems happy if only one is set.

So `__rvm_checksum_any && return 0` would be right (and the original `__rvm_checksum_none || return 0` too) but `_any && return` is probably readable?

Reply:
I changed to the equivalent `__rvm_checksum_any && return 0`.